db/migrate/20241021175734_create_preferences.rb

```diff
@@ -1,9 +1,10 @@
 class CreatePreferences < ActiveRecord::Migration[7.1]
   def change
     create_table :preferences do |t|
+      t.references :user, foreign_key: true, null: false
       t.string :name
```

Comment:
If you are validating `name` presence in the model, here would be nice to not allow null in `name` as well. wdyt?